platform/android/os_android.cpp

```diff
@@ -170,6 +170,8 @@ Error OS_Android::open_dynamic_library(const String p_path, void *&p_library_han
 		so_file_exists = false;
 	}
 
+	ERR_FAIL_COND_V(!FileAccess::exists(path), ERR_FILE_NOT_FOUND);
```

Comment:
FYI, this breaks dynamic library loading on Android since on that platform the original path may be inaccessible which requires moving the library around (see logic below).

Reply:
Reverting the change for Android in https://github.com/godotengine/godot/pull/86792

Reply:
Actually I reviewed this a bit fast, this may also fail on other platforms. So the whole PR might need to be reverted.

Reply:
Eep, sorry, guys! I thought it would be fine

Reply:
It looks like we could report a different error code if the file doesn't exist on Android too, it would just need to be handled later, I guess as an `else` on the 2nd `if (internal_so_file_exists)` check?

Reply:
So my worry here is that this function is currently only used for GDExtension, OpenXR (for Android), and .NET (using absolute paths), so we might be forgetting about the fact that dynamic libraries can be in `C:\Windows\System32` or `/usr/lib`/`/usr/lib64`.

We don't use it currently (Linux does a bunch of dynamic loading but doesn't seem to rely on this method, it uses `dlopen` directly), but in theory I would expect this function to succeed if I do `OS::get_singleton()->open_dynamic_library("libSDL2.so", sdl_handle);` on Linux to load `/usr/lib64/libSDL2.so` on my system (with `/usr/lib64` being part of my distro's default library paths).

While I feel we're treating it more as a `open_gdextension_library` currently, and only taking this use case into account.

Am I missing something?

CC @bruvzg

Reply:
Actually looking further at the `OS_Unix` implementation I see it's not the generic wrapper around `dlopen` I thought it would be. It does seem focused only on trying to load a library next to the binary or in the parent `../lib` folder if it exists.

So I guess the other checks added for Unix/Windows are ok. But there should likely be an explicit `_MSG` that refers back to the original `p_path` and not just the generic error that would mention the modified `path`.

Because currently trying to open `mylib.so` if it doesn't exist will fail with an error referring to `../lib/mylib.so`.

Reply:
Currently, (for GDExtension and, .NET) it is OK to assume we are using full path, but for something like https://github.com/godotengine/godot/pull/85741 it won't be, also GDExtension can try using it to load its dependencies.

Maybe we should check if path is absolute path and only do this check (and path substitution). So `open_dynamic_library("res://libSDL2.so")` will check do it and `open_dynamic_library("libSDL2.so")` will skip the checks and let system look for the lib.

Reply:
> FYI, this breaks dynamic library loading on Android since on that platform the original path may be inaccessible which requires moving the library around (see logic below).

For this, we probably should try substituting path for the expected one, like on macOS it's trying to replace executable folder with `../Frameworks/` if lib is not found.